src/FastConv/FastConv.cpp

```diff
@@ -10,23 +11,259 @@ CFastConv::~CFastConv( void )
     reset();
 }
 
-Error_t CFastConv::init(float *pfImpulseResponse, int iLengthOfIr, int iBlockLength /*= 8192*/, ConvCompMode_t eCompMode /*= kFreqDomain*/)
+Error_t CFastConv::init(float* pfImpulseResponse, int iLengthOfIr, int iBlockLength /*= 8192*/, ConvCompMode_t eCompMode /*= kFreqDomain*/)
 {
+    //set processing mode
+    if (eCompMode!= kTimeDomain and eCompMode!= kFreqDomain)
+    {
+        std::cout<< "ConvCompMode_t has to be either kTimeDomain or kFreqDomain";
+        return Error_t::kFunctionInvalidArgsError;
+    }
+    m_eCompMode = eCompMode;
+    
+    this->reset();
+
+    
+    m_iCirIdx = 0;
+    m_iLengthOfIr = iLengthOfIr;
+    
+    m_iBlockLength = iBlockLength;
+    m_pfInputBlock = new float [m_iBlockLength*2];
+    for (int i=0; i<m_iBlockLength*2; i++)
+        m_pfInputBlock[i] = 0.f;
+    
+    m_pfOutputBlock = new float [m_iBlockLength];
+    for (int i=0; i<m_iBlockLength; i++)
+        m_pfOutputBlock[i] = 0.f;
+    
+    m_pfBlockConvOuput = new float [m_iBlockLength*2];
+    for (int i=0; i<m_iBlockLength*2; i++)
+        m_pfBlockConvOuput[i] = 0.f;
+    
+    
+    //slice pfImpulseResponse into m_pfIRMatrix (iBlockNum, iBlockLength)
```

Comment:
Related to other comment about time-domain processing: you probably don't want to slice up the impulse response in time-domain mode.